File: server/router/opportunity.go
package router

import (
	"server/api/opportunity"

	"github.com/gin-gonic/gin"
)

// InitOpportunityRouter 初始化岗位机会相关路由
func InitOpportunityRouter(privateGroup *gin.RouterGroup, publicGroup *gin.RouterGroup, adminGroup *gin.RouterGroup) {
	PublicOpportunityRouter := publicGroup.Group("/api/opportunities")
	{
		PublicOpportunityRouter.GET("", opportunity.ListPublicOpportunities)
	}

	AdminOpportunityRouter := adminGroup.Group("/api/admin/opportunities")
	{
		AdminOpportunityRouter.GET("", opportunity.ListAdminOpportunities)
		AdminOpportunityRouter.POST("", opportunity.CreateOpportunity)
		AdminOpportunityRouter.POST("/batch", opportunity.BatchCreateOpportunities)
		AdminOpportunityRouter.PUT("/:id", opportunity.UpdateOpportunity)
		AdminOpportunityRouter.DELETE("/:id", opportunity.ArchiveOpportunity)
	}
}

File: web/src/api/opportunity.ts
import apiClient from './client';
import type { ApiResponse } from '@/types/global';
import type {
  JobOpportunity,
  OpportunityBatchCreateRequest,
  OpportunityListParams,
  OpportunityListResponse,
  OpportunityUpsertRequest,
} from '@/types/opportunity';

export const opportunityAPI = {
  getPublicOpportunities: (params?: OpportunityListParams): Promise<ApiResponse<OpportunityListResponse>> => {
    return apiClient.get('/api/opportunities', { params });
  },

  getAdminOpportunities: (params?: OpportunityListParams): Promise<ApiResponse<OpportunityListResponse>> => {
    return apiClient.get('/api/admin/opportunities', { params });
  },

  createOpportunity: (data: OpportunityUpsertRequest): Promise<ApiResponse<JobOpportunity>> => {
    return apiClient.post('/api/admin/opportunities', data);
  },

  batchCreateOpportunities: (data: OpportunityBatchCreateRequest): Promise<ApiResponse<JobOpportunity[]>> => {
    return apiClient.post('/api/admin/opportunities/batch', data);
  },

  updateOpportunity: (id: number, data: OpportunityUpsertRequest): Promise<ApiResponse<JobOpportunity>> => {
    return apiClient.put(`/api/admin/opportunities/${id}`, data);
  },

  archiveOpportunity: (id: number): Promise<ApiResponse<null>> => {
    return apiClient.delete(`/api/admin/opportunities/${id}`);
  },
};
